zou/app/services/auth_service.py

```diff
@@ -27,153 +31,136 @@ def check_auth(app, email, password):
     """
-    strategy = app.config["AUTH_STRATEGY"]
-    if strategy == "auth_local_classic":
-        user = local_auth_strategy(email, password, app)
-    elif strategy == "auth_local_no_password":
-        user = no_password_auth_strategy(email)
-    elif strategy == "auth_remote_ldap":
-        user = ldap_auth_strategy(email, password, app)
-    else:
-        raise NoAuthStrategyConfigured
-
-    if user is None:
-        app.logger.error("No user found for: %s" % email)
-        raise WrongUserException(email)
-
-    if not user.get("active", False):
-        raise UnactiveUserException(user["email"])
-
-    return user
-
-
-def check_credentials(email, password, app=None):
-    """
-    Check if given password and email match an user in database.
-    Password hash comparison is based on BCrypt.
-    """
+    if not email:
+        raise WrongUserException()
     try:
         person = persons_service.get_person_by_email(email, unsafe=True)
     except PersonNotFoundException:
         try:
             person = persons_service.get_person_by_desktop_login(email)
         except PersonNotFoundException:
-            if app is not None:
-                app.logger.error("Person not found: %s" % (email))
-            raise WrongPasswordException()
+            raise WrongUserException()
+
+    if not person.get("active", False):
+        raise UnactiveUserException()
+
+    login_failed_attemps = person["login_failed_attemps"]
+    if login_failed_attemps is None:
+        login_failed_attemps = 0
+    if (
+        login_failed_attemps >= 5
+        and date_helpers.get_datetime_from_string(person["last_login_failed"])
+        + timedelta(minutes=1)
+        > datetime.now()
+    ):
+        raise TooMuchLoginFailedAttemps()
 
+    strategy = app.config["AUTH_STRATEGY"]
     try:
-        password_hash = person["password"] or ""
-
-        if bcrypt.check_password_hash(password_hash, password):
-            return person
+        if strategy == "auth_local_classic":
+            user = local_auth_strategy(person, password, app)
+        elif strategy == "auth_local_no_password":
+            user = person
+        elif strategy == "auth_remote_ldap":
+            user = ldap_auth_strategy(person, password, app)
         else:
-            if app is not None:
-                app.logger.error(
-                    "Wrong password for person: %s" % person["full_name"]
-                )
-            raise WrongPasswordException()
-    except ValueError:
-        if app is not None:
-            app.logger.error("Wrong password for: %s" % person["full_name"])
+            raise NoAuthStrategyConfigured()
+    except WrongPasswordException:
+        persons_service.update_login_failed_attemps_person(
+            person["id"], login_failed_attemps + 1, datetime.now()
+        )
         raise WrongPasswordException()
 
+    persons_service.update_login_failed_attemps_person(person["id"], 0)
 
-def no_password_auth_strategy(email):
```

Comment:
Please do not remove the no password strategy.

Reply:
I have re-added a function for that but it just return the person given in parameter.